### PR TITLE
task5

### PR DESCRIPTION
**Pull request by SevRyb**
Implemented tests for major use cases.

### CODE REVIEW

**Review by PavloFedakNULP — CHANGES_REQUESTED** [review]
Please add screen with % of coverage by tests

**Comment by SevRyb** [comment]
In my case it is convenient to estimate coverage in used methods (also slots and signals). Qt testing framework does not support automatic coverage measurement, but we can do it by simple analysis . It is also worth adding that the test cases implemented in UTSwClient.cpp include implicit methods calls for testing its functionality. There are 13 test cases to cover all cases of possible uses of the application (you can see below).  I left custom widgets classes without testing, because to test methods, which draw something on screen - is nonsense in my application. 

image


The project consists of 4 classes (MainWindow, GameHost, GameBoard, GameButton). Below you can see total methods, slots and signals by particular class.

MainWindow - **1 method, 10 slots**
GameHost - **9 methods, 1 slot, 5 signals**
GameBoard - **3 methods, 1 slot**
GameButton - **4 methods, 1 signal**

**Total: 35**

My test cases cover all methods (also slots and signals) except listed below:

Methods:
void GameBoard::reset()
void GameBoard::load(const QString &cells)
void GameButton::setState(char state)
void GameButton::id()
Slots:
void GameBoard::onPlayerChoice(int cellId, signed char cellState)
Signals:
void GameButton::hit(int id)

**Total uncovered methods (also slots and signals): 6**

image


**To summarize, the coverage is 83%.**

**Review by PavloFedakNULP — APPROVED** [review]
